### ISSUE TITLE
Optional Date synchronizes to 1970-01-01 00:00:08 +0000 instead of nil

### ISSUE THREAD

**Issue by TadeasKriz** [realm/realm-cocoa#6112]
## Goals

We have a Realm object with an optional `Date` property (among other properties). We want to use the `nil` value.

## Expected Results

The second phone has the date properties equal to `nil`.

## Actual Results

The second phone has those date properties equal to `1970-01-01 00:00:08 +0000`.

The **Realm Studio** app shows those columns as `null` which is expected.

I tried to investigate as much as I could and it seems that `m_seconds` of the `realm::Timestamp` column has raw value of `\f8` or something like that.

## Steps to Reproduce

1. create instances of the object on one phone
2. sync them through Realm.io Sync
3. log in from another phone using the same user
4. list the objects you created after sync finishes on the second device

## Code Sample

```
import RealmSwift

@objcMembers
class RealmTask: Object {
    dynamic var id = UUID().uuidString
    dynamic var parentId: String = ""
    dynamic var text: String = ""
    dynamic var completed: Date? = nil
    let subtasks = List<RealmTask>()
    dynamic var dueDate: Date? = nil
    dynamic var repeatInterval = RealmOptional<TimeInterval>()
}
```

## Version of Realm and Tooling

Realm framework version: 3.13.1

Realm Object Server version: v3.19.0-cloud.2

iOS/OSX version: 12.1.4

```
ProductName:	Mac OS X
ProductVersion:	10.14
BuildVersion:	18A391

/Applications/Xcode.app/Contents/Developer
Xcode 10.1
Build version 10B61

/usr/local/bin/pod
1.7.0.beta.2
(not in use here)

/bin/bash
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)

/usr/bin/git
git version 2.17.2 (Apple Git-113)
```

**Comment by brianzinken** [comment]
I am also seeing this behavior.  Data synced from through MSSQL adapter to Realm Cloud in a consolidated database.  Realm cloud shows null dates, but the phones  are showing  1970-01-01 00:00:08 +0000.

**Comment by tgoyne** [comment]
I can't reproduce this. I've tried with both query-based sync and full sync and it seems to be working fine.

**Comment by brianzinken** [comment]
If it helps, it used to work for me. I have tried changing the realm version  for my project starting with 3.14 and working backwards to 3.10 without any success.

**Comment by brianzinken** [comment]
I have a reference realm that has a partial realm with nulls from March 21st.  Other partial realms for the same reference realm from a day or two later have the 1970-01-01 00:00:08 +0000 value instead.  Please let me know if there is any specific data that I can provide to help.

**Comment by tgoyne** [comment]
The server team has tracked down a bug that may be responsible for this. The problematic case is apparently just when creating a new partial view on the server for an existing reference realm that already contains `nil` values for a Timestamp property. I don't have an ETA yet for when the fix will be deployed.

**Comment by ironage** [comment]
This was fixed in https://github.com/realm/realm-sync/pull/2894 and released in sync 4.2.1 which is ROS version v3.20.0-rc.1. Deployed to cloud as raas v801, on April 5 2019.
